README.md

```diff
@@ -0,0 +1,378 @@
+The northbound API will by default listen at 0.0.0.0:7070, and the southbound will by default listen at 0.0.0.0:9090. Those IP addresses and ports are user configurable.
+
+You can also run it as a Linux service. Install the service:
+
+    sudo ./install
+
+And then run the service:
+
+    sudo service on-static start
+
+The log files are located at:
+
+    /var/log/on-static/on-static.log
+
+
+## Deploy
+
+It's recommend to doploy on-static on a standalone host to offload the host that runs RackHD.
+
+The host running on-static better to have two NICs, one for northbond API and one for southbond API. The NIC for southbound API should be accessible by RackHD nodes. 
+One example is to connect the northbond NIC to RachHD control network.
+
+![Deploy Example](on-static-deploy.bmp)
+
+## Use it with RackHD
+
+Assuming user already have a RackHD instance running and he/she wantted to install ubuntu to one of his/her nodes. 
+
+First, it will has to setup a unbutu image server. The setups are:
+
+1. Locate the unbutu iso file from somewhere. User can choose to download it by themselves or just ask on-static to download it instead. 
+2. Setup the ubuntu image server by send a PUT request to on-static northbound API, something look like:
+
+        curl -X PUT "http://10.62.59.150:7070/images?name=ubuntu&version=14.04&isoweb=http://10.62.59.150:9090/iso/photon-1.0.iso"
+
+3. Make sure your RackHD config file has the following configurations so that it will go to an external static file server for os images.
+
+    ```
+    "fileServerAddress": "172.31.128.4", # this is the northbond IP of on-static, make sure it's accessible by managed nodes. 
+    "fileServerPort": 9090,
+    "fileServerPath": "/",
+    ```
+
+    RackHD configure files can be find at:
+
+        /opt/monorail/config.json
+    or
+
+        /opt/onrack/etc/monorail.json
+
+4. Issue OS install workflow with repo key in payload set to "http://on-static-ip-addr:port/ubuntu/14.04". 
+
+    The API look like:
+
+        http://{{host}}/api/2.0/nodes/:identifier/workflows?name=Graph.InstallUbuntu
+
+    And the payload look like:
+
+    ```
+    {
+        "options": {
+            "defaults": {
+                "version": "trusty",
+                "baseUrl": "install/netboot/ubuntu-installer/amd64",
+                "kargs": {
+                    "live-installer/net-image": "http://on-static-ip-addr:port/ubuntu/14.04/ubuntu/install/filesystem.squashfs"
+                },
+                "repo": "http://on-static-ip-addr:port/ubuntu/14.04"
+            }
+        }
+    }
+    ```
+
+## API
+
+### Northbound
+
+* Image management
+
+    1. GET http://0.0.0.0:7070/images: Get/list all OS images. No parameter is needed.
+
+        ```
+        curl http://0.0.0.0:7070/images | python -m json.tool
+        [
+            {
+                "id": "6807e9b2-a763-4b74-bfe9-4b20fe964400",
+                "iso": "client.iso",
+                "name": "photon",
+                "status": "OK",
```

Comment:
what does  status 'OK' mean?  file image is ok, or repo is prepared. just a questions at first glance